plugins/modules/zypper.py

```diff
@@ -511,8 +511,25 @@ def repo_refresh(m):
     return retvals
 
 
+def contains_readonly_opt(opts):
+    return re.match('^ro,|,ro,|,ro$|^ro$', opts) is not None
+
+
+def get_fs_type_and_readonly_state(mount_point):
+    with open('/proc/mounts', 'r') as file:
+        for line in file.readlines():
+            fields = line.split()
+            path = fields[1]
+            if path == mount_point:
+                fs = fields[2]
+                opts = fields[3]
+                return fs, contains_readonly_opt(opts)
+    return None
+
+
 def transactional_updates():
-    return os.path.exists('/var/lib/misc/transactional-update.state')
+    return get_fs_type_and_readonly_state('/') == ('btrfs', True) and os.path.exists('/usr/sbin/transactional-update')
```

Comment:
I would first check for existence of `/usr/sbin/transactional-update`, since in most cases where transactional updates are not used this will likely already be `False`:
```suggestion
    return os.path.exists('/usr/sbin/transactional-update') and get_fs_type_and_readonly_state('/') == ('btrfs', True)
```